mongoose.c

```diff
@@ -8120,6 +8110,80 @@ bool mg_span(struct mg_str s, struct mg_str *a, struct mg_str *b, char sep) {
   }
 }
 
+size_t mg_str_to_num_(struct mg_str str, int base, void *val, size_t val_len);
+size_t mg_str_to_num_(struct mg_str str, int base, void *val, size_t val_len) {
+  size_t i = 0, ndigits = 0;
+  uint64_t max = val_len == 1   ? 0xFF
+                 : val_len == 2 ? 0xFFFF
+                 : val_len == 4 ? 0xFFFFFFFF
+                                : (uint64_t) ~0;
+  uint64_t max2 = max / 10, result = 0;
+  while (i < str.len && (str.buf[i] == ' ' || str.buf[i] == '\t')) i++;
+  if (base == 0 && str.len >= 2) {
+    if (str.buf[i] == '0') {
+      i++;
+      base = str.buf[i] == 'b' ? 2 : str.buf[i] == 'x' ? 16 : 10;
```

Comment:
strtol logic assumes numbers that start with `0` but followed by digits to be octal.
Shall we employ the same logic?
i.e.

```c
 base = str.buf[i] == 'b' ? 2 : str.buf[i] == 'x' ? 16 : 8;
```